nima/http2/webserver/src/test/java/io/helidon/nima/http2/webserver/ConnectionConfigTest.java

```diff
@@ -0,0 +1,182 @@
+import io.helidon.nima.http2.Http2Setting;
+import io.helidon.nima.http2.Http2Settings;
+import io.helidon.nima.webserver.ConnectionContext;
+import io.helidon.nima.webserver.Router;
+import io.helidon.nima.webserver.Routing;
+import io.helidon.nima.webserver.WebServer;
+import io.helidon.nima.webserver.http.DirectHandlers;
+import io.helidon.nima.webserver.spi.ServerConnectionProvider;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class ConnectionConfigTest {
+
+    // ConnectionContext mockup
+    private static class TestContext implements ConnectionContext {
+
+        @Override
+        public PeerInfo remotePeer() {
+            return null;
+        }
+
+        @Override
+        public PeerInfo localPeer() {
+            return null;
+        }
+
+        @Override
+        public boolean isSecure() {
+            return false;
+        }
+
+        @Override
+        public String socketId() {
+            return null;
+        }
+
+        @Override
+        public String childSocketId() {
+            return null;
+        }
+
+        @Override
+        public MediaContext mediaContext() {
+            return null;
+        }
+
+        @Override
+        public ContentEncodingContext contentEncodingContext() {
+            return null;
+        }
+
+        @Override
+        public ExecutorService sharedExecutor() {
+            return null;
+        }
+
+        @Override
+        public DataWriter dataWriter() {
+            return null;
+        }
+
+        @Override
+        public DataReader dataReader() {
+            return null;
+        }
+
+        @Override
+        public Router router() {
+            return new Router() {
+                @Override
+                public <T extends Routing> T routing(Class<T> routingType, T defaultValue) {
+                    return null;
+                }
+
+                @Override
+                public void afterStop() {
+                }
+
+                @Override
+                public void beforeStart() {
+                }
+            };
+        }
+
+        @Override
+        public long maxPayloadSize() {
+            return 0;
+        }
+
+        @Override
+        public DirectHandlers directHandlers() {
+            return null;
+        }
+
+        @Override
+        public Context context() {
+            return null;
+        }
+    }
+
+    @Test
+    void testConnectionConfig()
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
+
+        // This will pick up application.yaml from the classpath as default configuration file
+        Config config = Config.create();
+
+        // Builds LoomServer instance including connectionProviders list.
+        WebServer.Builder wsBuilder = WebServer.builder()
+                .config(config.get("server"));
```

Comment:
If you want to check that the configuration was propagated, you can have static methods on your custom connection provider, build the server, and then check that your connection provider obtained the correct setup.

Reply:
Well, yes, that's 1st part of the test. But it's not enough.
I would also like to know that those specific 2 values described in the issue got propagated into local Http2Settings instance of the Http2Connection.